src/homework6/modelfromlife.py

```diff
@@ -0,0 +1,275 @@
+            bets = {}
+            for person in self.roulette_players:
+                if person.do_i_have_cash():
+                    bets.setdefault(person.bet_roulette())
+            self.all_bets_are_off(bets)
+            bets.clear()
+        self.roulette_players.clear()
+
+    def all_bets_are_off(self, bets):
+        # It is a very simple simulation roulette. Only numbers
+        # played in this simulation. Guests can't place bet for a
+        # zero because a casino always win.
+        roulette_round = random.choice(self.roulette_table)
+        for person in bets:
+            if person[1][0] == roulette_round:
+                self.capital -= (person[1][1] * 35)
+                person[0].won(person[1][1] * 36)
+            else:
+                self.capital += person[1][1]
+
+    def one_hand_bandit(self, person):
+        if person.do_i_have_cash():
+            bet = person.bet_one_hand()
+            win_ratio = 1
+            random.shuffle(self.one_hand_bandit_)
+            column_1 = self.one_hand_bandit_[2:5]
+            random.shuffle(self.one_hand_bandit_)
+            column_2 = self.one_hand_bandit_[2:5]
+            random.shuffle(self.one_hand_bandit_)
+            column_3 = self.one_hand_bandit_[2:5]
+            if column_1[0] == column_2[0] == column_3[0]:
+                win_ratio += 1
+            if column_1[1] == column_2[1] == column_3[1]:
+                if column_1[1] == column_2[1] == column_3[1] == 7:
+                    win_ratio += 3
+                else:
+                    win_ratio += 1
+            if column_1[2] == column_2[2] == column_3[2]:
+                win_ratio += 1
+            if win_ratio > 1:
+                self.capital -= bet * (win_ratio - 1)
+                person.won(bet * win_ratio)
+            else:
+                self.capital += bet
+
+    def blackjack(self, persons_list):
+        # It simulation does't use different ace values.
+        cards = self.blackjack_deck[:] * 4
+        random.shuffle(cards)
+        players = []
+        croupier_hand = [cards.pop(0), cards.pop(1)]
+        for person in persons_list:
+            if person.do_i_have_cash():
+                bet = person.bet_blackjack()
+                hand = [cards.pop(0), cards.pop(1)]
+                one_more = person.get_hand(hand)
+                players.append([person, bet, hand, one_more])
+        while any(param[-1] for param in players):
+            for person in players:
+                if person[-1]:
+                    person[2].append(cards.pop(0))
+                    person[-1] = person[0].get_hand(person[2])
+        while sum(croupier_hand) < 17:
+            croupier_hand.append(cards.pop(0))
+        for person in players:
+            if 22 > sum(person[2]) > sum(croupier_hand):
+                person[0].won(person[1] * 2)
+                self.capital -= person[1]
+            else:
+                self.capital += person[1]
+
+    def bar(self):
```

Comment:
bar can be put to separate class and here it would be an attribute

Reply:
separated